File: questApplication/src/main/java/ru/nekit/android/qls/quest/answer/common/IAnswerChecker.java
package ru.nekit.android.qls.quest.answer.common;

import android.support.annotation.NonNull;

import ru.nekit.android.qls.quest.Quest;

public interface IAnswerChecker<T> {

    boolean checkAlternativeInput(@NonNull Quest quest, @NonNull T answer);

    boolean checkStringInputFormat(@NonNull Quest quest, @NonNull String value);

    boolean checkStringInput(@NonNull Quest quest, @NonNull String value);

}

File: questDomain/src/main/java/ru/nekit/android/qls/domain/answerChecker/common/IAnswerChecker.kt
package ru.nekit.android.qls.domain.answerChecker.common

import ru.nekit.android.qls.domain.model.quest.Quest

interface IAnswerChecker<in R : Quest> {

    fun checkAlternativeInput(quest: R, answer: Any): Boolean

    fun checkStringInputFormat(quest: R, value: String): Boolean

    fun checkStringInput(quest: R, value: String): Boolean

}
